### DISCUSSION TITLE
Request headers change using intercept

### DISCUSSION THREAD

**Discussion by GrzegorzBednarski** [cypress-io/cypress#16092] [Questions and Help]
Hi
I want to remove some items after test, using request. Unfortunately I need to pass Authorization param inside request headers.
So I want to use request but can't pass headers using cy.request.
I read that i should use intercept (as I understand I should use it before request to change request values. 
I tried :
cy.intercept({
        method: 'DELETE',
        url: `/api/credentials/${credentialID}`
      },
      {
        headers: {
          "Authorization": getToken()
        }
      }
  );
  cy.request('DELETE', `/api/credentials/${credentialID}`);
});

Unfortunately Cypress didn't add my Authorization value to headers.
Any suggestions what I'm doing wrong? Using Cypress 7.0.1

**Comment by GrzegorzBednarski** [comment]
Also tried :
  cy.intercept(`/api/credentials/${credentialID}`, (req) => {
    req.headers['Authorization'] = getToken();
  })
  cy.request('DELETE', `/api/credentials/${credentialID}`);
});
With no success

**Comment by bahmutov** [comment]
If you are trying to pass additional headers using `cy.request` command, take a look at https://glebbahmutov.com/blog/cypress-request-and-cookies/

Also, remember that `cy.intercept` does NOT "see" or intercept the HTTP calls made using `cy.request`.

**Comment by GrzegorzBednarski** [comment]
@bahmutov  thanks. 
It helps. 
Please add headers into documentation - request syntax -> https://docs.cypress.io/api/commands/request#Syntax
